test/pummel/test-http-upload-timeout.js

```diff
@@ -28,14 +28,17 @@ const http = require('http');
 const server = http.createServer();
 let connections = 0;
 
+function ontimeout() {
+  throw new Error('Unexpected timeout');
+}
+
 server.on('request', function(req, res) {
   req.socket.setTimeout(1000);
-  req.socket.on('timeout', function() {
-    throw new Error('Unexpected timeout');
-  });
+  req.socket.on('timeout', ontimeout);
   req.on('end', function() {
     connections--;
     res.writeHead(200);
+    req.socket.off('timeout', ontimeout);
```

Comment:
Why is this necessary? Why do inactivity timeouts fire after `'end'` is emitted? Is it because of https://github.com/nodejs/node/pull/43522?

Reply:
I think it will make memory leak. If the `i` in `for` change to 100, we can see the warning as follows.
```
(node:31736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 timeout listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:31736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 timeout listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
(node:31736) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 timeout listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
```